### PR TITLE
feat: update function adn status codes

### PR DESCRIPTION
**Pull request by muneebkq**
add missing function and error codes

### CODE REVIEW

File: examples/LexTypes.h
Comment (ahmad-kemsan) [inline examples/LexTypes.h:1]: @muneebkq Please refer to the latest LexTypes.h. There were some changes made recently. Please update this file as well.

File: examples/LexTypes.h [hunk 1]
```diff
@@ -3,6 +3,7 @@
 
 #define BUFFER_SIZE 256
```
Comment (ahmad-kemsan) [inline examples/LexTypes.h]: @muneebkq Please refer to the latest types.h file. It seems you missed including a header file `stdint.h`. This file is needed for older version of gcc as we are using int64_t.

**Review by ahmad-kemsan — CHANGES_REQUESTED** [review]
Please check the comments

File: examples/LexTypes.h (hunk 1 above)
Comment (muneebkq) [reply examples/LexTypes.h]: refactored

**Review by ahmad-kemsan — APPROVED** [review]

**Pull request merged**